velox/functions/sparksql/MapFromEntries.cpp

```diff
@@ -0,0 +1,230 @@
+          }
+
+          // Check nested null in keys.
+          if (keyVector->containsNullAt(keyIndex)) {
+            resetSize(row);
+            VELOX_USER_FAIL(fmt::format(
+                "{}: {}",
+                kIndeterminateKeyErrorMessage,
+                keyVector->toString(keyIndex)));
+          }
+        }
+      });
+    }
+
+    context.deselectErrors(*remianingRows.get());
+
+    VectorPtr wrappedKeys;
+    VectorPtr wrappedValues;
+    if (decodedRowVector->isIdentityMapping()) {
+      wrappedKeys = rowVector->childAt(0);
+      wrappedValues = rowVector->childAt(1);
+    } else if (decodedRowVector->isConstantMapping()) {
+      if (decodedRowVector->isNullAt(0)) {
+        // If top level row is null, child might not be addressable at index 0
+        // so we do not try to read it.
+        wrappedKeys = BaseVector::createNullConstant(
+            rowVector->childAt(0)->type(),
+            decodedRowVector->size(),
+            context.pool());
+        wrappedValues = BaseVector::createNullConstant(
+            rowVector->childAt(1)->type(),
+            decodedRowVector->size(),
+            context.pool());
+      } else {
+        wrappedKeys = BaseVector::wrapInConstant(
+            decodedRowVector->size(),
+            decodedRowVector->index(0),
+            rowVector->childAt(0));
+        wrappedValues = BaseVector::wrapInConstant(
+            decodedRowVector->size(),
+            decodedRowVector->index(0),
+            rowVector->childAt(1));
+      }
+    } else {
+      // Dictionary.
+      auto indices = allocateIndices(decodedRowVector->size(), context.pool());
+      memcpy(
+          indices->asMutable<vector_size_t>(),
+          decodedRowVector->indices(),
+          BaseVector::byteSize<vector_size_t>(decodedRowVector->size()));
+      // Any null in the top row(X, Y) should be marked as null since its
+      // not guranteed to be addressable at X or Y.
+      for (auto i = 0; i < decodedRowVector->size(); i++) {
+        if (decodedRowVector->isNullAt(i)) {
+          bits::setNull(mutableNulls, i);
+        }
+      }
+      wrappedKeys = BaseVector::wrapInDictionary(
+          nulls, indices, decodedRowVector->size(), rowVector->childAt(0));
+      wrappedValues = BaseVector::wrapInDictionary(
+          nulls, indices, decodedRowVector->size(), rowVector->childAt(1));
+    }
+
+    // To avoid creating new buffers, we try to reuse the input's buffers
+    // as many as possible.
+    auto mapVector = std::make_shared<MapVector>(
+        context.pool(),
+        outputType,
+        nulls,
+        rows.end(),
+        inputArray->offsets(),
+        sizes,
+        wrappedKeys,
+        wrappedValues);
+
+    checkDuplicateKeys(mapVector, *remianingRows, context);
+    return mapVector;
+  }
+};
+} // namespace
+
+VELOX_DECLARE_VECTOR_FUNCTION(
+    udf_map_from_entries,
+    MapFromEntriesFunction::signatures(),
+    std::make_unique<MapFromEntriesFunction>());
```

Comment:
I'm not sure if it makes sense to move MapFromEntriesFunction of presotsql to lib and reuse it in sparksql, as the two implementations look similar. A template parameter could be introduced to control different behaviors on null.

Reply:
A property `allowNullEle_`  is introduced to function for distinguishing Presto and Spark behavior.